### PR TITLE
[Query Generator] simplify IN, NOT IN criteria text

### PR DESCRIPTION
**Pull request by faissaloux**
Hellooo 👋🏻 

### Current
We need to pass criteria text with quotes for each value to generate a valid query.

image

### After

image

### CODE REVIEW

File: js/src/functions.js [hunk 1]
```diff
@@ -441,6 +441,15 @@ Functions.escapeSingleQuote = function (s) {
     return s.replaceAll('\\', '\\\\').replaceAll('\'', '\\\'');
 };
 
+/**
+ * @param {string} s
+ * @param {string} char
+ * @return {string}
+ */
+Functions.wrap = function (s, char) {
```
Comment (MauricioFauth) [inline js/src/functions.js:449]: If this function is only used in a single place, it doesn't need to be in functions.js.

**Review by MauricioFauth — CHANGES_REQUESTED** [review]
Does it still accepts using single quotes? What about a single value that has a comma?
I'm not sure if this should go to QA_5_2.

**Comment by MauricioFauth** [comment]
> No it does not, if using single quotes it escapes them like if they belong to the value like any other character.

> This won't work because it counts as two separate values. Do you have any idea to solve this?

These are breaking changes.

The only way to accept not needing a quoted value is to also accept quoted values, as a quoted value fixes the comma issue. But accepting both ways could be very confusing.

I think a good solution is to only accept quoted values and add a button to edit the list, like the enum/set editor that exists when creating/editing a table.

**Comment by faissaloux** [comment]
> I think a good solution is to only accept quoted values and add a button to edit the list, like the enum/set editor that exists when creating/editing a table.

Yeah this is the kind of the solution I thought about, but for addition too.

We need to be able to add values one by one (without quotes, quotes can be part of the value) and then can be edited or removed there on a list of values.

**Comment by faissaloux** [comment]
@MauricioFauth

https://github.com/user-attachments/assets/166fd993-f5f0-417b-9476-1cd8b5707c84

**Comment by MauricioFauth** [comment]
That's nice!

**Review by MauricioFauth — APPROVED** [review]
Great work! Thank you!

Could you please change the target branch to `master` instead of `QA_5_2` and squash your commits?

**Comment by faissaloux** [comment]
@MauricioFauth

Retargeted to `master` and commits squashed!

**Pull request merged**